cubelib/src/cube/cube_edges.rs

```diff
@@ -225,6 +227,78 @@ impl Default for CenterEdgeCube {
     }
 }
 
+impl CenterEdgeCube {
+    #[inline]
+    #[cfg(target_feature = "avx2")]
+    pub fn random(parity: bool, rng: &mut ThreadRng) -> Self {
+        let bytes = random_edges(parity, rng);
+        unsafe { avx2::unsafe_from_bytes(bytes) }
+    }
+
+    #[inline]
+    #[cfg(all(target_arch = "wasm32", not(target_feature = "avx2")))]
+    pub fn random(parity: bool, rng: &mut ThreadRng) -> Self {
+        let bytes = random_edges(parity, rng);
+        wasm32::from_bytes(bytes)
+    }
+
+    #[inline]
+    #[cfg(all(target_feature = "neon", not(target_feature = "avx2")))]
+    pub fn random(parity: bool, rng: &mut ThreadRng) -> Self {
+        let bytes = random_edges(parity, rng);
+        unsafe { neon::unsafe_from_bytes(bytes) }
+    }
+
+}
+
+fn random_edges(parity: bool, rng: &mut ThreadRng) -> [u8; 12] {
+    let mut edge_bytes: [u8; 12] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
+    let mut orientation_parity = false;
+    let mut swap_parity = false;
+
+    fn get_bytes(position_id: u8, piece_id: u8, flipped: bool) -> u8 {
```

Comment:
Figured out the edge orientation encoding. Requires some logic to get the right bits